### ISSUE TITLE
Unmatched shape in DNet

### ISSUE THREAD

**Issue by lhao0301** [XPixelGroup/HYPIR#24]
https://github.com/XPixelGroup/HYPIR/blob/56d0ddfce20905172fad832396bc920d149b7d55/HYPIR/model/D.py#L55-L88

The `channel num` of output features in image encoder are [ 768, 1536, 1024]. However the decoder needs [384, 768, 1536].

It may be some problems in function `_visual_forward`?
https://github.com/XPixelGroup/HYPIR/blob/56d0ddfce20905172fad832396bc920d149b7d55/HYPIR/model/backbone.py#L7-L30

**Comment by 0x3f3f3f3fun** [comment]
The version of open_clip_torch or timm might be incorrect.

**Comment by lhao0301** [comment]
Thanks for your advice. The problem was solved after seting the same version for `timm` and `open_clip_torch` as list in `requirements.txt`.

Met another problem,  the `out` for DNet is very large, with the demo code as follows,

```
import torch
from src.models.hypir_D import ImageConvNextDiscriminator
import timm
import open_clip
print('timm version: ', timm.__version__)
print('open_clip version: ', open_clip.__version__)

DNet = ImageConvNextDiscriminator(precision="fp32").to(device='cuda:0')
DNet.eval().requires_grad_(False)
data = torch.randn((1, 3, 400, 512)).to('cuda:0')
data = torch.clip(data, -1, 1)
out = DNet(data, for_G=True)
print(out) 

# timm version:  1.0.15
# open_clip version:  2.31.0
# tensor([[14813.8057]], device='cuda:0')
```

**Comment by 0x3f3f3f3fun** [comment]
This is ok. The output of discriminator (loss) is very large at the beginning of training, it will decrease to a normal range quickly.
